tempodb/encoding/vparquet/block_traceql.go

```diff
@@ -295,6 +296,12 @@ func (i *spansetIterator) Next() (*span, error) {
 			}
 		}
 
+		// spans returned from the filter are not guaranteed to be in file order
+		// we need them to be so that the meta iterators work correctly. sort here
+		sort.Slice(i.currentSpans, func(j, k int) bool {
+			return parquetquery.CompareRowNumbers(DefinitionLevelResourceSpans, i.currentSpans[k].rowNum, i.currentSpans[j].rowNum) == 1
```

Comment:
Is this different than `compare(j,k) == -1` ?

Reply:
i'm not sure what you mean. is there a faster way to do this?

Reply:
oh. understood. i agree that the suggested ordering is more clear.